pkg/manager/member/tiproxy_member_manager.go

```diff
@@ -260,13 +259,19 @@ func (m *tiproxyMemberManager) syncStatus(tc *v1alpha1.TidbCluster, sts *apps.St
 	}
 
 	pods := helper.GetPodOrdinals(tc.Status.TiProxy.StatefulSet.Replicas, sts)
-	members := make(map[string]bool)
+	members := make(map[string]v1alpha1.TiProxyMember)
 	for id := range pods {
-		if _, err := m.deps.ProxyControl.GetConfigProxy(tc, id); err != nil {
-			klog.Infof("failed to get tiproxy[%d] config: %+v", id, err)
-			continue
+		name := fmt.Sprintf("%s-%d", controller.TiProxyMemberName(tc.GetName()), id)
+		health := true
+		if err := m.deps.ProxyControl.IsHealth(tc, id); err != nil {
+			klog.V(4).Infof("tiproxy[%d] is not health: %+v", id, err)
+			health = false
+		}
+		members[name] = v1alpha1.TiProxyMember{
+			Name:            name,
+			Health:          health,
+			HealthCheckTime: metav1.Now(),
```

Comment:
`HealthCheckTime` shall not be updated on every sync, or it will trigger informer and requeue tc(cause tc to be synced forever).